### PR TITLE
Don’t dive into null parts of AST.

### PR DESCRIPTION
**Pull request by davidungar**
When compiling code with certain errors, it is possible to get null pointers in the AST.
The compiler indexes even when there are errors, and the indexing code then tries to walk into the null pointers, causing a crash. As a result, the frontend crashes, and the developer does not get to see the errors. This PR adds a null check for this specific case. It does not solve the general problem.

Resolves rdar://42314665

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
@akyrtzi Did you want to look at this? Thanks!

File: lib/AST/USRGeneration.cpp [hunk 1]
```diff
@@ -226,7 +226,7 @@ bool ide::printDeclUSR(const ValueDecl *D, raw_ostream &OS) {
     return printObjCUSR(D, OS);
   }
 
-  if (!D->hasInterfaceType())
+  if (!D->hasInterfaceType() || D->getInterfaceType()->is<ErrorType>())
```
Comment (akyrtzi) [inline lib/AST/USRGeneration.cpp]: This is the reason the test is filing, could you verify whether this change is necessary, is it really fixing a crash in your original test case ?

**Comment by akyrtzi** [comment]
Generally it would be ideal if you could add a regression test but I can understand if it is too difficult to reduce the original failing test case.

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Review by akyrtzi — APPROVED** [review]

**Comment by davidungar** [comment]
I agree a regression test would be great. If I can come up with one, I'll do another PR.

**Pull request merged**